File: AddressBookAPI/Models/Core/ContactModel.cs
//using Microsoft.Build.Framework;
using System.ComponentModel.DataAnnotations;

namespace AddressBookAPI.Models.CoreModels
{
    public class Contact
    {
        public int? Id { get; set; }

        [Required(ErrorMessage = "Name is Required"), MinLength(5, ErrorMessage = "min length is 5 chars")]
        public string Name { get; set; }

        [Required(ErrorMessage = "Email is Required"), EmailAddress(ErrorMessage = "Email is invalid")]
        public string Email { get; set; }

        [Required(ErrorMessage = "Mobile is Required"), Phone(ErrorMessage = "Mobile is invalid")]
        public string Mobile { get; set; }

        public string? Landline { get; set; }

        [Url(ErrorMessage = "Invalid URL")]
        public string? website { get; set; }

        public Address? Address { get; set; }

    }
}

File: Address Book/TS/models/contactModel.ts
import { Address } from "./addressModel";
export class Contact{
    id:number;
    name: string;
    email: string;
    mobile: string;
    landline: string | null;
    website: string | null;
    address: Address;

    constructor(name: string, email: string, mobile: string, landline: string|null, website: string|null, address: Address,id:number) {

        this.id = id;
        this.name = name;
        this.email = email;
        this.mobile = mobile;
        this.landline = landline;
        this.website = website;
        this.address = address;
    }
}